### PR TITLE
Run command with hot-reload

### PR DESCRIPTION
**Pull request by ShivanshuKGupta**
Changes:
1. Added a run command
2. Hot-reload

### CODE REVIEW

File: packages/gazelle_cli/lib/commands/run/run_project.dart [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+      /// Else send the input to the process, the process will be
+      /// responsible for if the event is 'r' then reload the code
+      process?.stdin.writeln(event);
+    }
+  });
+
+  ProcessSignal.sigint.watch().listen((event) {
+    process?.kill();
+
+    /// TODO: Delete the temporary directory here after the process is killed
```
Comment (filippomenchini) [inline packages/gazelle_cli/lib/commands/run/run_project.dart:160]: Is this a left over?

Comment (ShivanshuKGupta) [reply packages/gazelle_cli/lib/commands/run/run_project.dart:160]: Deleting the tmp project directory might not be necessary right now, but, we might need it to in the future. That is why I left it as it is.

**Review by filippomenchini — APPROVED** [review]
LGTM!

File: packages/gazelle_cli/lib/commands/run/run_project.dart (hunk 1 above)
Comment (filippomenchini) [reply packages/gazelle_cli/lib/commands/run/run_project.dart]: Removed TODO with 5dd54963a21867b1cb4952fe85bd97fd1b645507

**Pull request merged**